shell/browser/browser.cc

```diff
@@ -26,6 +26,23 @@
 
 namespace electron {
 
+namespace {
+
+// Call |quit| after Chromium is fully started.
+//
+// This is important for quitting immediately in the "ready" event, when
+// certain initialization task may still be pending, and quitting at that time
+// could end up with crash on exit.
+void RunQuitClosure(base::OnceClosure quit) {
+  // Consume remaining tasks in message loop.
+  base::RunLoop().RunUntilIdle();
+  // On Linux/Windows the "ready" event is emitted in "PreMainMessageLoopRun",
+  // make sure we quit after message loop has run for once.
+  base::ThreadTaskRunnerHandle::Get()->PostTask(FROM_HERE, std::move(quit));
```

Comment:
If we're calling `PostTask`, do we also need to be calling `RunUntilIdle`?

Reply:
Hmm `RunUntilIdle` does seem unnecessary here, let me try without it.